pkg/jivavolume/jivavolume.go

```diff
@@ -102,16 +150,89 @@ func HasResourceParameters(req *csi.CreateVolumeRequest) ResourceParameters {
 	}
 }
 
-// WithReplicaStorageClass returns storage class
-func WithReplicaStorageClass(sc string) string {
-	if sc == "" {
-		return "openebs-hostpath"
-	}
-	return sc
-}
-
 // WithSpec defines the Spec field of JivaVolume
 func (j *Jiva) WithSpec(spec jv.JivaVolumeSpec) *Jiva {
 	j.jvObj.Spec = spec
 	return j
 }
+
+// WithPV defines the PV field of JivaVolumeSpec
+func (j *Jiva) WithPV(pvName string) *Jiva {
+	j.jvObj.Spec.PV = pvName
+	return j
+}
+
+// WithCapacity defines the Capacity field of JivaVolumeSpec
+func (j *Jiva) WithCapacity(capacity string) *Jiva {
+	j.jvObj.Spec.Capacity = capacity
+	return j
+}
+
+// WithReplicaSC defines the ReplicaSC field of JivaVolumePolicySpec
+func (j *Jiva) WithReplicaSC(scName string) *Jiva {
+	if scName == "" {
+		scName = defaultReplicaSC
+	}
+	j.jvObj.Spec.Policy.ReplicaSC = scName
+	return j
+}
+
+// WithEnableBufio defines the ReplicaSC field of JivaVolumePolicySpec
+func (j *Jiva) WithEnableBufio(enable bool) *Jiva {
+	j.jvObj.Spec.Policy.EnableBufio = enable
+	return j
+}
+
+// WithAutoScaling defines the ReplicaSC field of JivaVolumePolicySpec
+func (j *Jiva) WithAutoScaling(enable bool) *Jiva {
+	j.jvObj.Spec.Policy.AutoScaling = enable
+	return j
+}
+
+// WithTarget defines the ReplicaSC field of JivaVolumePolicySpec
+func (j *Jiva) WithTarget(target jv.TargetSpec) *Jiva {
+	if target.ReplicationFactor == 0 {
+		target.ReplicationFactor = defaultTargetSpec.ReplicationFactor
+	}
+	if target.AuxResources == nil {
+		target.AuxResources = defaultTargetSpec.AuxResources
+	}
+	if target.Resources == nil {
+		target.Resources = defaultTargetSpec.Resources
+	}
+	if target.Tolerations == nil {
+		target.Tolerations = defaultTargetSpec.Tolerations
+	}
+	if target.Affinity == nil {
+		target.Affinity = defaultTargetSpec.Affinity
+	}
+	if target.NodeSelector == nil {
+		target.NodeSelector = defaultTargetSpec.NodeSelector
+	}
+	if target.PriorityClassName == "" {
+		target.PriorityClassName = defaultTargetSpec.PriorityClassName
+	}
+	j.jvObj.Spec.Policy.Target = target
```

Comment:
Optional: can we mv entire if cases to a map and then iterate over it to set default values ?